### DISCUSSION TITLE
New driver Ember and old Sonoff ZBBridge wifi version

### DISCUSSION THREAD

**Discussion by JackTalisker** [Koenkk/zigbee2mqtt#23418] [Devices]
I've been using for a few years - without particular issues - a Sonoff ZBBridge (the older version) as described here: https://www.zigbee2mqtt.io/advanced/remote-adapter/connect_to_a_remote_sonoff_zbbridge.html

This is the coordinator : https://zigbee.blakadder.com/Sonoff_ZBBridge.html
The wifi side is esp8266 based (tasmota flashed) while the zigbee side is EFR32MG21 based (flashed with 6.7.9.0 build 405 firmware).

Since zigbee2mqt ver.1.39.0 is telling me that the ezsp driver will be definitively replaced with the ember driver in next release, the question is straight forward.
Will I be able to use that new driver with my coordinator?

Thank you

**Comment by francisp2** [comment]
There is new firmware here:

https://github.com/xsp1989/zigbeeFirmware/tree/master

**Reply by JackTalisker** [reply]
So I have to flash the SM-011 7.4.x firmware to make the coordinator compatible with the ember driver?

**Reply by francisp2** [reply]
indeed

**Reply by JackTalisker** [reply]
Thank you francis

**Reply by JackTalisker** [reply]
the switch from ezsp to ember version went without problems.
Thank you.

**Reply by davidrpfarinha** [reply]
Thanks for the info, @francisp2. 
Do you know any reason why they didn't submit this firmware to the main Tasmota branch?
I was already contemplating purchasing a new coordinator, but it looks like I can maintain this for a little longer, and do a through research.

**Reply by francisp2** [reply]
Because the new firmware works with Zigbee2MQTT and ZHA, but does NOT work with Zigbee2Tasmota.

**Reply by davidrpfarinha** [reply]
> Because the new firmware works with Zigbee2MQTT and ZHA, but does NOT work with Zigbee2Tasmota.

Ok, thanks for the clarification. 

> But wouldn't following the above mentioned documentation by Jack, would lead you to use Z2T? Or am I read it wrong?
> Meaning, my ZBBridge is flashed with Tasmota, and being used as the coordinator for my Z2M. Just got myself confused with this. 😄 

I think I got it 🤦‍♂️ The Z2T is like what I have for ZHA where I don't have a broker to receive, and process the messages, as it is the Sonoff ZBBridge Pro doing all that.

**Reply by milothomas** [reply]
> the switch from ezsp to ember version went without problems. Thank you.

Hey @JackTalisker, glad you got it working. Do you have links to a guide on how this was done? 

I followed the guide from Digiblur years ago and flashed a the `ncp-uart...ota` file using a TTL and then also flashed a Tasmota firmware using Tasmotizer. 

With this new firmware, do I flash this using the TTL? And if so, do I still flash the Tasmota firmware?

**Reply by JackTalisker** [reply]
> > the switch from ezsp to ember version went without problems. Thank you.
> 
> Hey @JackTalisker, glad you got it working. Do you have links to a guide on how this was done?
> 
> I followed the guide from Digiblur years ago and flashed a the `ncp-uart...ota` file using a TTL and then also flashed a Tasmota firmware using Tasmotizer.
> 
> With this new firmware, do I flash this using the TTL? And if so, do I still flash the Tasmota firmware?

You can use OTA update for flashing.
In my case I simply switched to Module 75 (according to Digiblur's guide), flashed the new updated OTA file version via Tasmota gui and then gone back to Module 0 (again with Digiblur's command )

**Reply by milothomas** [reply]
> > > the switch from ezsp to ember version went without problems. Thank you.
> > 
> > 
> > Hey @JackTalisker, glad you got it working. Do you have links to a guide on how this was done?
> > I followed the guide from Digiblur years ago and flashed a the `ncp-uart...ota` file using a TTL and then also flashed a Tasmota firmware using Tasmotizer.
> > With this new firmware, do I flash this using the TTL? And if so, do I still flash the Tasmota firmware?
> 
> You can use OTA update for flashing. In my case I simply switched to Module 75 (according to Digiblur's guide), flashed the new updated OTA file version via Tasmota gui and then gone back to Module 0 (again with Digiblur's command )

Thanks, I was able to update the firmware.

**Reply by devzwf** [reply]
Hi @JackTalisker  , @milothomas 

sorry to necro this ,  but when you said : 
> flashed the new updated OTA file version via Tasmota gui

you mean this FW :  https://github.com/xsp1989/zigbeeFirmware/tree/master/firmware/ZigbeeBridge_SM-011-signed
via theTasmota " Upgrade by file upload "

then switch back to modu 0 via :
`backlog rule1 on system#boot do TCPStart 8888 endon ; rule1 1 ; template {"NAME":"Sonoff ZHABridge","GPIO":[56,208,0,209,59,58,0,0,0,0,0,0,17],"FLAG":0,"BASE":18} ; module 0`

just want to make sure i am not missing anything , as unfortunetly for me the device is remote

Note: planning a trip there next year to replace the ZBbridge with a dongle or an smlight

thanks in advance

**Reply by JackTalisker** [reply]
My steps to upgrade from firmware 6.7.9 to 7.4.3.0 where:
Downloaded ncp-uart-sw_7.4.3.0_115200.ota 
Main menu > Configuration > Configure module > selected "Module 75" > Save
Main menu > firmware upgrade > Upgrade by file upload > selected previously downloaded file "ncp-uart-sw_7.4.3.0_115200.ota" > Start upgrade
Main menu > Tools > Console > Entered command: "backlog rule1 on system#boot do TCPStart 8888 endon ; rule1 1 ; template {"NAME":"Sonoff ZHABridge","GPIO":[56,208,0,209,59,58,0,0,0,0,0,0,17],"FLAG":0,"BASE":18} ; module 0" (without quotes) and hit Return

In my case I was able to upgrade the fw flawlessly. 
I hope you will be able as well but I'm not responsible of any problem.

**Reply by devzwf** [reply]
Thanks 
no worry about dammage or problem :)

**Reply by ximirafe** [reply]
It works!! You guys saved my zigbee network. Thank you all!!

**Reply by peter--s** [reply]
Note that changing NCP using Tasmota from anything newer than the officially supported EZSP v8 requires Tasmota [version 14.1.0](http://ota.tasmota.com/tasmota/release-14.1.0/tasmota-zbbridge.bin.gz) or earlyer according to https://github.com/arendst/Tasmota/issues/23001.

**Comment by acca84** [comment]
Hello, I'd like to switch my sonoff zb bridge from ezsp to ember driver.
I'm using it with zigbee2mqtt 
You can see my bridge informations on the image below.
I'd like to avoid bricking my device.
Is that possible to downgrade my tasmota version to 14.1.0 in order to flash my device as advised by JackTalisker ?
Thanks
<img width="1220" height="4148" alt="Screenshot_2026-09-17-22-14-57-062_com android chrome-edit" src="https://github.com/user-attachments/assets/1ea370bf-81fd-485c-adda-087c384fdc38" />
